llvm/lib/Passes/StandardInstrumentations.cpp

```diff
@@ -2493,6 +2551,79 @@ void PrintCrashIRInstrumentation::registerCallbacks(
       });
 }
 
+void InstrCountChangedReporter::registerCallbacks(
+    PassInstrumentationCallbacks &PIC, ModuleAnalysisManager &MAM) {
+  PIC.registerBeforeNonSkippedPassCallback([this, &MAM, Registered = false](
```

Comment:
is it possible to know ahead of time if remarks are enabled so we can avoid calling these callbacks after every pass in the common case where we don't want remarks?

Reply:
It is controlled by `pass-remarks-analysis` option in `lib/IR/DiagnosticHandler.cpp`, and return value comes from a virtual method `isAnalysisRemarkEnabled`. Currently no derived class in codebase overrides this method.

Reply:
I see `ClangDiagnosticHandler` overrides this in `clang/lib/CodeGen/CodeGenAction.cpp`

Reply:
`Module::shouldEmitInstrCountChangedRemark()` calls `getContext().getDiagHandlerPtr()->isAnalysisRemarkEnabled("size-info")`, we should probably do something similar and call `Context.getDiagHandlerPtr()->isAnalysisRemarkEnabled("size-info")`

Reply:
Oh, I see the `LLVMContext` in constructor of `StandardInstrumentations`.